fedimint-server-ui/src/setup.rs

```diff
@@ -147,9 +181,12 @@ async fn setup_submit(
 }
 
 // GET handler for the /login route (display the login form)
-async fn login_form(State(state): State<UiState<DynSetupApi>>) -> impl IntoResponse {
-    if state.api.setup_code().await.is_none() {
-        return Redirect::to(ROOT_ROUTE).into_response();
+async fn login_form(
+    State(state): State<UiState<DynSetupApi>>,
+    maybe_auth: MaybeUserAuth,
+) -> impl IntoResponse {
+    if state.api.auth().await.is_none() || maybe_auth.is_authenticated() {
```

Comment:
If we need to do it in every handler, we should just move it to the extractor itself. It does have access to `state`, and can do anything.